### PR TITLE
[Numpy]  add numpy op bitwise_xor, hsplit, moveaxis, rot90

### PR DESCRIPTION
**Pull request by gyshi**
add numpy op bitwise_xor

fix some format problems

add numpy op moveaxis

fix code style

add numpy op rot90

fix pylint error

updata rot90

address comments

add  numpy op hsplit

refactor split compute function

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -677,3 +677,50 @@ def _np_trace(a, offset=0, axis1=0, axis2=1, out=None):
...
+
+def moveaxis(a, source, destination):
+    """Move axes of an array to new positions.
+
+    Other axes remain in their original order.
+
+    Parameters
+    ----------
+    a : ndarray
+    The array whose axes should be reordered.
```
Comment (sxjscience) [inline python/mxnet/_numpy_op_doc.py]: Add spaces here.

Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: can you have find other problems , i will fix them together

Comment (sxjscience) [reply python/mxnet/_numpy_op_doc.py]: Sure, I'm reviewing this file.

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -2012,6 +2012,121 @@ def split(ary, indices_or_sections, axis=0):
...
+        section_size = int(axis_size / sections)
+        indices = [i * section_size for i in range(sections)]
+    elif isinstance(indices_or_sections, (list, set, tuple)):
+        indices = [0] + list(indices_or_sections)
+    else:
+        raise ValueError('indices_or_sections must either int or tuple of ints')
+    ret = _npi.hsplit(ary, indices, axis, False)
+    if not isinstance(ret, list):
+        raise NotImplementedError('single output from hsplit is not supported yet...')
+    return ret
```
Comment (sxjscience) [inline python/mxnet/ndarray/numpy/_op.py]: Why do we need hsplit given that we have implemented split?

File: src/operator/numpy/np_matrix_op.cc [hunk 3]
```diff
@@ -612,5 +614,215 @@ NNVM_REGISTER_OP(_backward_npi_flip)
...
+    if (axes[i] < 0) {
+      for (int j = 0; j < axes.ndim(); ++j) {
+        if (state_axes[j] == false) {
+          axes[i] = j;
+          state_axes[j] = true;
+          break;
+        }
+      }
+    }
+  }
```
Comment (sxjscience) [inline src/operator/numpy/np_matrix_op.cc]: Consider to refactor the code to reduce redundant logic.

**Comment by sxjscience** [comment]
LGTM in general, for implementing the forward op based on the forward/backward of another op, you may refer to the PR in https://github.com/apache/incubator-mxnet/pull/12821

File: python/mxnet/ndarray/numpy/_op.py (hunk 2 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: beacause numpy has this op， and we provide this op can make users more friendly

Comment (sxjscience) [reply python/mxnet/ndarray/numpy/_op.py]: I mean, we can directly implement hsplit as `split(axis=1)`

Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: I  know what you mean, but in numpy, when your input data is 1 dim, the axis will be  0, if ndim > 1, axis will be 1.

File: src/operator/numpy/np_matrix_op.cc (hunk 3 above)
Comment (gyshi) [reply src/operator/numpy/np_matrix_op.cc]: i think it is not redundant logic, because i want to record historical information.

Comment (gyshi) [reply src/operator/numpy/np_matrix_op.cc]: you can see numpy moveaxis,  you must give origin corresponding order

Comment (sxjscience) [reply src/operator/numpy/np_matrix_op.cc]: I mean, the code also appears in NumpyMoveaxisCompute

File: python/mxnet/ndarray/numpy/_op.py (hunk 2 above)
Comment (sxjscience) [reply python/mxnet/ndarray/numpy/_op.py]: I see. @reminisce, @gyshi , I think we need to think about how to refactor the code. Some common operators should share the same node, this helps simplify the graph and would be extremely helpful for graph fusion.

File: src/operator/numpy/np_matrix_op.cc (hunk 3 above)
Comment (sxjscience) [reply src/operator/numpy/np_matrix_op.cc]: @gyshi Would you take a look at the issue?

**Comment by gyshi** [comment]
@sxjscience  i refactor this code
